ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -90,8 +94,23 @@ private String getPartitionName(Table table) throws HiveException {
   }
 
   private CompactionResponse compact(Table table, String partitionName) throws HiveException {
-    CompactionResponse resp = context.getDb().compact2(table.getDbName(), table.getTableName(), partitionName,
-        desc.getCompactionType(), desc.getProperties());
+    CompactionType compactionType = null;
+    String strType = desc.getCompactionType();
+    if ("major".equalsIgnoreCase(strType)) {
+      compactionType = CompactionType.MAJOR;
+    } else if ("minor".equalsIgnoreCase(strType)) {
+      compactionType = CompactionType.MINOR;
+    } else {
+      throw new RuntimeException("Unknown compaction type " + strType);
+    }
+
+    CompactionRequest req = new CompactionRequest(table.getDbName(), table.getTableName(), compactionType);
+    req.setPartitionname(partitionName);
+    req.setPoolName(desc.getPoolName());
+    req.setProperties(desc.getProperties());
+    req.setInitiatorId(JavaUtils.hostname() + "-" + HiveMetaStoreClient.MANUALLY_INITIATED_COMPACTION);
+    req.setInitiatorVersion(HiveMetaStoreClient.class.getPackage().getImplementationVersion());
+    CompactionResponse resp = context.getDb().compact3(req);
```

Comment:
can we rename the `compact3` -> `compact`?

Reply:
The code contains both compact and compact2 methods. For backward compatibility reasons, I created a new method instead of chanigng the signature of an existing one.

Reply:
you can just overload the method. New one is accepting CompactionRequest object